core/services/functions/external_adapter_client.go

```diff
@@ -0,0 +1,130 @@
+package functions
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type ExternalAdapterClient struct {
+	AdapterURL       url.URL
+	MaxResponseBytes int64
+}
+
+type secretsPayload struct {
+	Endpoint  string      `json:"endpoint"`
+	RequestId string      `json:"requestId"`
+	JobName   string      `json:"jobName"`
+	Data      secretsData `json:"data"`
+}
+
+type secretsData struct {
+	RequestType          string `json:"requestType"`
+	EncryptedSecretsUrls string `json:"encryptedSecretsUrls"`
+}
+
+type response struct {
+	Result     string        `json:"result"`
+	Data       *responseData `json:"data"`
+	StatusCode int           `json:"statusCode"`
+}
+
+type responseData struct {
+	Result      string `json:"result"`
+	Error       string `json:"error"`
+	ErrorString string `json:"errorString"`
+}
+
+func (ea ExternalAdapterClient) FetchEncryptedSecrets(ctx context.Context, encryptedSecretsUrls []byte, requestId string, jobName string) (encryptedSecrets, userError []byte, err error) {
+	encodedSecretsUrls := base64.StdEncoding.EncodeToString(encryptedSecretsUrls)
+
+	data := secretsData{
+		RequestType:          "fetchThresholdEncryptedSecrets",
+		EncryptedSecretsUrls: encodedSecretsUrls,
+	}
+
+	payload := secretsPayload{
+		Endpoint:  "fetcher",
+		RequestId: requestId,
+		JobName:   jobName,
+		Data:      data,
+	}
+
+	encryptedSecrets, userError, err = ea.request(ctx, payload, requestId, jobName)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error fetching encrypted secrets")
+	}
+
+	return encryptedSecrets, userError, nil
+}
+
+func (ea ExternalAdapterClient) request(
+	ctx context.Context,
+	payload interface{},
+	requestId string,
+	jobName string,
+) (userResult, userError []byte, err error) {
```

Comment:
my bad, I forgot that []byte applies to both but I'm surprised linter isn't complaining